plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -457,24 +461,24 @@ private List<Slice> rewriteFile(String sourcePath, FileDeletion deletion)
         try {
             String tablePath = rootTableLocation.toString();
             Location sourceLocation = Location.of(sourcePath);
-            String sourceRelativePath = relativePath(tablePath, sourcePath);
+            String sourceReferencePath = getReferencedPath(tablePath, sourcePath);
 
             Location targetLocation = sourceLocation.sibling(session.getQueryId() + "_" + randomUUID());
-            String targetRelativePath = relativePath(tablePath, targetLocation.toString());
+            String targetReferencePath = getReferencedPath(tablePath, targetLocation.toString());
```

Comment:
Will this rewrite file in base table in case of CLONE?

Reply:
You are right, thanks for pointing out! I changed the logic, cloned table files should now under the cloned table location, PTAL